functorch/functorch/_src/config.py

```diff
@@ -20,6 +20,9 @@
 #   fix for complex numbers
 use_fake_tensor = False
 
+# Changes AOTAutograd to passing a list of tensors that are then cleared
```

Comment:
The user is passing the list?

Reply:
So today, the contract with the compiler is that we pass in a function like
```
f(a, b, c)
```

and the compiler returns a callable
```
compiled_f(a,b,c)
```

This changes the contract so that was pass in a function
```
f([a,b,c])
```
and the compiler returns a callable

```
compiled_f([a,b,c])
```

Reply:
I'm not sure to see the difference? Don't you use pytree to "unpack" any data structure in the args?

Reply:
@albanD I'll add some more context to this issue/commit, but basically, it's about object lifetimes.

If you call `f(a,b,c)`, then a, b, and c will stay alive for the duration of `f`, and there's no way around this AFAIK.

OTOH, if you call `f([a,b,c])`, then we can clear the list and free a/b/c inside of `f`.

Reply:
You can do `del a, b, c` ?

Reply:
Unfortunately not :( Gimme a sec.